src/auth/validate.rs

```diff
@@ -0,0 +1,108 @@
+use crate::auth::flows::AuthProvider;
+use crate::auth::get_user_from_pat;
+use crate::auth::AuthenticationError;
+use crate::database::models::user_item;
+use crate::models::users::{Role, User, UserId, UserPayoutData};
+use actix_web::http::header::HeaderMap;
+use reqwest::header::{HeaderValue, AUTHORIZATION};
+
+pub async fn get_user_from_headers<'a, E>(
+    headers: &HeaderMap,
+    executor: E,
+    redis: &deadpool_redis::Pool,
+) -> Result<User, AuthenticationError>
+where
+    E: sqlx::Executor<'a, Database = sqlx::Postgres> + Copy,
+{
+    let token: Option<&HeaderValue> = headers.get(AUTHORIZATION);
+
+    // Fetch DB user record and minos user from headers
+    let db_user = get_user_record_from_bearer_token(
```

Comment:
Cookie implementation is still TODO, right? Using the bearer token as the primary authentication point is totally good but we should definitely have a way to get the mra_ bearer token from a cookie

Reply:
we won't support cookie auth due to csrf, the frontend supports this auth method